judge/views/user.py

```diff
@@ -335,6 +338,59 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class WCIPEGMergeActivate(LoginRequiredMixin, TitleMixin, FormView):
+    template_name = 'user/wcipeg-merge-activate.html'
+    form_class = WCIPEGMergeActivationForm
+
+    def merge(self, from_user, to_user):
+        from_user.merge_user = to_user
+        from_user.save()
+        with transaction.atomic():
+            Submission.objects.filter(user=from_user).update(user=to_user)
+            Comment.objects.filter(user=from_user).update(user=to_user)
+            CommentVote.objects.filter(voter=from_user).update(voter=to_user)
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            token = 'wcipeg_%d_%s' % (int(kwargs['pk']), request.user.email)
```

Comment:
Is an `int` cast necessary here?

Reply:
I may be wrong, but I don't think `(?P<pk>\d+)` would automatically convert to `int`.

Reply:
We don't need it as an `int`, though. It's getting formatted as part of a string anyways. I guess we can keep it if it makes it more clear that `pk` should be an integer.

Reply:
Although I can do:
```suggestion
            token = 'wcipeg_%s_%s' % (kwargs['pk'], request.user.email)
```
if that's what you mean.

Reply:
Yes